.github/workflows/ci-cd.yml

```
@@ -0,0 +1,302 @@
+    - name: 🧪 Run API integration tests
+      run: yarn test:api --exit-on-failure
+      env:
+        API_BASE_URL: http://localhost:3001
+
+    - name: 📊 Run performance benchmarks
+      run: yarn test:api:benchmarks
+      env:
+        API_BASE_URL: http://localhost:3001
+
+    - name: 📄 Upload test results
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.10)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)